File: EntityFrameworkCore.Diagrams/EntityFrameworkCore.Diagrams.Frontend/src/app/models/db-entity-property.ts
import { ClrType } from './clr-type';
import { ValueGenerated } from './value-generated';

export class DbEntityProperty {
    name: string;
    clrType: ClrType;
    isConcurrencyToken: boolean;
    isNullable: boolean;
    isReadOnlyAfterSave: boolean;
    isReadOnlyBeforeSave: boolean;
    isShadowProperty: boolean;
    isStoreGeneratedAlways: boolean;
    requiresValueGenerator: boolean;
    valueGenerated: ValueGenerated;

    static fromJSON(obj: Object): DbEntityProperty {
        return Object.assign(new DbEntityProperty(), obj, {
            clrType: ClrType.fromJSON(obj['clrType'])
        });
    }

    equals(other: DbEntityProperty): boolean {
        return other instanceof DbEntityProperty && (
            this === other
            || this.name === other.name && this.clrType.equals(other.clrType)
        );
    }
}

File: EntityFrameworkCore.Diagrams/EntityFrameworkCore.Diagrams/Dto/DbEntityProperty.cs
using Microsoft.EntityFrameworkCore.Metadata;

namespace EntityFrameworkCore.Diagrams.Dto
{
    public class DbEntityProperty
    {
        public string Name { get; set; }

        public ClrType ClrType { get; set; }

        public bool IsConcurrencyToken { get; internal set; }

        public bool IsNullable { get; internal set; }

        public bool IsReadOnlyAfterSave { get; internal set; }

        public bool IsReadOnlyBeforeSave { get; internal set; }

        public bool IsShadowProperty { get; internal set; }

        public bool IsStoreGeneratedAlways { get; internal set; }

        public bool RequiresValueGenerator { get; internal set; }

        public ValueGenerated ValueGenerated { get; internal set; }
    }
}